docs/developers-guide.md

```diff
@@ -1,8 +1,190 @@
 # Developers Guide
 
+
+
 ## Extending the GraphDB Workbench
 
+## Plugin system
+
+Since v9.1, GraphDB workbench features a plugin system which allows components to be plugged in 
+without introducing coupling between new and existing components. The new system allows extending or 
+replacing existing components, introduction of new single or compositional components. All this 
+could be achieved without any changes in the rest of the system.
+
+_Currently the new system is integrated in the workbench main components registration. These are the
+components which implement the main workbench views (extension point `route`) and their respective 
+main menu entries (extension point `main.menu`). In next versions more extension points might be 
+introduced._
+
+### What is the plugin system and how can a developer can use it?
+
+The plugin system consist of three components. A `PluginRegistry` which is a service that has a role 
+to maintain a runtime registry of all registered to given extension point plugins. It has the 
+following interface:
+```javascript
+PluginRegistry {
+    add(extensionPoint:String, pluginDefinition:[Object|Array]),
+    get(extensionPoint:String):Array,
+    clear(extensionPoint:String),
+    listModules()
+}
+```
+The second component is the `plugin.js` file which the developer defines for each extension point
+where he needs new behavior or customization.  
+The last one is the `plugins.js` file which is autogenerated and is composed from all `plugin.js`
```

Comment:
Maybe mention it's generated by Webpack ?

Reply:
It's described in the following sections